docs/userguide/quickstart.txt

```diff
@@ -2,211 +2,189 @@
 ``setuptools`` Quickstart
 ==========================
 
+.. contents::
+
 Installation
 ============
 
-.. _Installing Packages: https://packaging.python.org/tutorials/installing-packages/
-
 To install the latest version of setuptools, use::
 
     pip install --upgrade setuptools
 
-Refer to `Installing Packages`_ guide for more information.
+
+Python packaging at a glance
+============================
+The landscape of Python packaging is shifting and ``Setuptools`` has evolved to
+only provide backend support, no longer being the de-facto packaging tool in
+the market. All python package must provide a ``pyproject.toml`` and specify
+the backend (build system) it wants to use. The distribution can then
+be generated with whatever tools that provides a ``build sdist``-alike
+functionality. While this may appear cumbersome, given the added pieces,
+it in fact tremendously enhances the portability of your package. The
+change is driven under `PEP 517 <https://www.python.org/dev/peps/pep-0517/#
+build-requirements>``. To learn more about Python packaging in general,
+navigate to the `bottom <Resources on python packaging>`_ of this page.
+
 
 Basic Use
 =========
+For basic use of setuptools, you will need a ``pyproject.toml`` with the
+exact following info, which declares you want to use ``setuptools`` to
+package your project:
+
+.. code-block:: toml
+
+    [build-system]
+    requires = ["setuptools", "wheel"]
+    build-backend = "setuptools.build_meta"
+
+Then, you will need a ``setup.cfg`` to specify your package information,
+such as metadata, contents, dependencies, etc. Here we demonstrate the minimum
+
+.. code-block:: ini
+
+    [metadata]
+    name = "mypackage"
+    version = 0.0.1
 
-For basic use of setuptools, just import things from setuptools.  Here's a
-minimal setup script using setuptools::
+    [options]
+    packages = "mypackage"
+    install_requires =
+      requests
+      importlib; python_version == "2.6"
 
-    from setuptools import setup, find_packages
-    setup(
-        name="HelloWorld",
-        version="0.1",
-        packages=find_packages(),
-    )
+This is what your project would look like::
 
-As you can see, it doesn't take much to use setuptools in a project.
-Run that script in your project folder, alongside the Python packages
-you have developed.
+    ~/mypackage/
+        pyproject.toml
+        setup.cfg
+        mypackage/__init__.py
 
-Invoke that script to produce distributions and automatically include all
-packages in the directory where the setup.py lives.  See the `Command
-Reference`_ section below to see what commands you can give to this setup
-script. For example, to produce a source distribution, simply invoke::
+Then, you need an installer, such as `pep517 <https://pypi.org/project/pep517/>`_
+which you can obtain via ``pip install pep517``. After downloading it, invoke
+the installer::
 
-    setup.py sdist
+    pep517 build
```

Comment:
Last I checked, the only way to invoke `pep517` is with `python -m pep517.build`.

Reply:
my mistake it should be ``python -m pep517.build``

Reply:
please let me know the part that looks iffy and I will push a new commit to update them:)